--- a/node_modules/twitter-openapi-typescript-generated/dist/models/Entities.js
+++ b/node_modules/twitter-openapi-typescript-generated/dist/models/Entities.js
@@ -45,7 +45,7 @@
         'media': json['media'] == null ? undefined : (json['media'].map(Media_1.MediaFromJSON)),
         'symbols': json['symbols'],
         'timestamps': json['timestamps'] == null ? undefined : (json['timestamps'].map(Timestamp_1.TimestampFromJSON)),
-        'urls': (json['urls'].map(Url_1.UrlFromJSON)),
+        'urls': json['urls'] == null ? undefined : (json['urls'].map(Url_1.UrlFromJSON)),
         'userMentions': json['user_mentions'],
     };
 }
@@ -64,7 +64,7 @@
         'media': value['media'] == null ? undefined : (value['media'].map(Media_1.MediaToJSON)),
         'symbols': value['symbols'],
         'timestamps': value['timestamps'] == null ? undefined : (value['timestamps'].map(Timestamp_1.TimestampToJSON)),
-        'urls': (value['urls'].map(Url_1.UrlToJSON)),
+        'urls': value['urls'] == null ? undefined : (value['urls'].map(Url_1.UrlToJSON)),
         'user_mentions': value['userMentions'],
     };
 }

--- a/node_modules/twitter-openapi-typescript-generated/dist/models/ExtendedEntities.js
+++ b/node_modules/twitter-openapi-typescript-generated/dist/models/ExtendedEntities.js
@@ -33,7 +33,7 @@
         return json;
     }
     return {
-        'media': (json['media'].map(MediaExtended_1.MediaExtendedFromJSON)),
+        'media': json['media'] == null ? undefined : (json['media'].map(MediaExtended_1.MediaExtendedFromJSON)),
     };
 }
 exports.ExtendedEntitiesFromJSONTyped = ExtendedEntitiesFromJSONTyped;
@@ -47,7 +47,7 @@
         return value;
     }
     return {
-        'media': (value['media'].map(MediaExtended_1.MediaExtendedToJSON)),
+        'media': value['media'] == null ? undefined : (value['media'].map(MediaExtended_1.MediaExtendedToJSON)),
     };
 }
 exports.ExtendedEntitiesToJSONTyped = ExtendedEntitiesToJSONTyped;
